Source/options.js

```diff
@@ -358,10 +358,17 @@ $(function() {
   }
 
   function loadStorageAndUpdate(callback) {
+    storage.clear();
     storage.get(null, function(data) {
       console.log('data: ' + data + ' : ' + JSON.stringify(data));
       var d = {};
-      if (!data || JSON.stringify(data) == '{}') { // in this case, storage was not initialized yet
+      data = {activation:true};
+      /*
+      * In this case, we have to check for storage not initialized
+      * as well as check if the object data has fields other than
+      * activation or not.
+      **/
+      if (!data || JSON.stringify(data) == '{}' || !data.savedPatterns) {
```

Comment:
I don't understand these changes.

In line 365 your setting data to {activation:true}. Therefore, the condition in line 371 will always be true. Consequently, the stored data will always be overwritten by the default in line 374.  This seems wrong.